frontend/src/components/ViewActivities/index.js

```diff
@@ -8,12 +15,77 @@ import ActivityList from './activitylist.js';
  * - `tripId` {string} The trip's ID. This is sent to the component through the URL. 
  */
 class ViewActivities extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    this.state = { addingActivity : false }
+  }
+
+  /**
+   * Complete "Add Activity" operation
+   */
+  addActivity = (e) => {
+    e.preventDefault();
+    this.setState({ addingActivity: true });
+
+    const newData = this.emptyActivity(this.props.match.params.tripId);
+    this.setState({ newAct: newData });
+  }
+
+  /**
+   * Runs when the user is done adding an activity.
+   */
+  doneAddingActivity = () => { 
+    this.setState({ 
+      addingActivity: false,
+      newAct: null 
+    }); 
+  };
+
+  /**
+   * Create an empty activity (with filler information) to edit and then display.
+   * Allows us to use editActivity instead of creating a whole new form for it. 
+   * 
+   * @param {string} tripId The tripId to attach to this new activity.
+   */
+  emptyActivity = (tripId) => {
```

Comment:
I would rename this to be more clear that you are returning an activity. Something like `getEmptyActivity` or `createEmptyActivity`.